code/game/machinery/autolathe.dm

```diff
@@ -59,171 +35,195 @@
-		matching_designs.Cut()
+		if(!istext(design_id))
+			return
 
-		for(var/v in stored_research.researched_designs)
-			var/datum/design/D = SSresearch.techweb_design_by_id(v)
-			if(findtext(D.name,params["to_search"]))
-				matching_designs.Add(D)
-		. = TRUE
+		if(!stored_research.isDesignResearchedID(design_id))
+			return
 
-	if(action == "make")
-		if (!busy)
-			/////////////////
-			//href protection
-			being_built = stored_research.isDesignResearchedID(params["id"])
-			if(!being_built)
-				return
-
-			var/multiplier = text2num(params["multiplier"])
-			if(!multiplier || !IS_FINITE(multiplier))
-				stack_trace("Invalid multiplier value in stack creation [multiplier], [usr] is likely attempting an exploit")
-				return
-			var/is_stack = ispath(being_built.build_path, /obj/item/stack)
-			multiplier = clamp(round(multiplier),1,50)
-
-			/////////////////
-
-			var/coeff = (is_stack ? 1 : creation_efficiency) //stacks are unaffected by production coefficient
-			var/total_amount = 0
-
-			for(var/MAT in being_built.materials)
-				total_amount += being_built.materials[MAT]
-
-			var/power = max(active_power_usage, (total_amount)*multiplier/5) //Change this to use all materials
-
-			var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
-
-			var/list/materials_used = list()
-			var/list/custom_materials = list() //These will apply their material effect, This should usually only be one.
-
-			for(var/MAT in being_built.materials)
-				var/datum/material/used_material = MAT
-				var/amount_needed = being_built.materials[MAT] * coeff * multiplier
-				if(istext(used_material)) //This means its a category
-					var/list/list_to_show = list()
-					for(var/i in SSmaterials.materials_by_category[used_material])
-						if(materials.materials[i] > 0)
-							list_to_show += i
-
-					used_material = tgui_input_list(usr, "Choose [used_material]", "Custom Material", sort_list(list_to_show, /proc/cmp_typepaths_asc))
-					if(isnull(used_material))
-						return //Didn't pick any material, so you can't build shit either.
-					custom_materials[used_material] += amount_needed
-
-				materials_used[used_material] = amount_needed
-
-			if(materials.has_materials(materials_used))
-				busy = TRUE
-				to_chat(usr, span_notice("You print [multiplier] item(s) from the [src]"))
-				use_power(power)
-				icon_state = "autolathe_n"
-				var/time = is_stack ? 32 : (32 * coeff * multiplier) ** 0.8
-				addtimer(CALLBACK(src, PROC_REF(make_item), power, materials_used, custom_materials, multiplier, coeff, is_stack, usr), time)
-				. = TRUE
-			else
-				to_chat(usr, span_alert("Not enough materials for this operation."))
-		else
+		var/datum/design/design = SSresearch.techweb_design_by_id(design_id)
+
+		if(!(design.build_type & AUTOLATHE) || design.id != design_id)
+			return
+
+		if (busy)
 			to_chat(usr, span_alert("The autolathe is busy. Please wait for completion of previous operation."))
```

Comment:
does it use this? line 233 seems to be very similar

Reply:
Yes, it does. It's to stop spamming the lathe, cause creating new in-world atoms is surprisingly expensive, especially when you have to move them. See the many, **many** issues TG and SR have with empty bottles flooding the halls and murdering the server for an example of why this is needed.